shell/browser/api/frame_subscriber.cc

```diff
@@ -72,11 +72,16 @@ void FrameSubscriber::RenderViewDeleted(content::RenderViewHost* host) {
   }
 }
 
-void FrameSubscriber::RenderViewHostChanged(content::RenderViewHost* old_host,
-                                            content::RenderViewHost* new_host) {
-  if ((old_host && old_host->GetWidget() == host_) || (!old_host && !host_)) {
+void FrameSubscriber::RenderFrameHostChanged(
```

Comment:
We should also remove `RenderViewHost` usage here https://github.com/electron/electron/blob/dfe501941cd9e7092829c3ef8e27e322047cf736/shell/browser/api/frame_subscriber.cc#L30, replace instead with `RenderFrameHost` of the primary frame.

Also swap `RenderViewDeleted` -> `RenderFrameDeleted` https://github.com/electron/electron/blob/dfe501941cd9e7092829c3ef8e27e322047cf736/shell/browser/api/frame_subscriber.cc#L69

Reply:
Also given `old_host` is not used in the following method, can this instead use `PrimaryPageChanged`

Reply:
Updated in https://github.com/electron/electron/pull/36820/commits/0f863161de29d0f13b0cb08b1c848c1c46862656